### ISSUE TITLE
Does not obey metered WiFi restrictions

### ISSUE THREAD

**Issue by achilleas-k** [syncthing/syncthing-android#760]
I'm having the exact same issue as reported in #409.
I am connected to a WiFi network that is marked as metered, SyncThing is set to run only when charging and on WiFi, yet it still runs when I plug it in.
### Version Information

```
App Version: 0.8.6
Syncthing Version: v0.14.8
Android Version: Android 6.0.1.
```

**Comment by Nutomic** [comment]
I just tried this, and it works most of the time. But apparently, it doesn't respect the setting in some cases. It would help if you can find steps to reproduce this problem.

**Comment by achilleas-k** [comment]
Interesting. I'll see what I can do.

**Comment by asayanna** [comment]
Hi. Think this might be related... 
I have Sync only on WiFi set on. I also have Restrict to certain WiFi Networks set. Syncthing just consumed 850mb of my mobile data. 
Syncthing 0.14.8
Syncthing Android version 0.8.6 
Android version 6.0.1

**Comment by Nutomic** [comment]
This should be fixed in 0.10.0. If you still have this problem, please open a new issue with logs.
